src/schema/definition.rs

```diff
@@ -1,25 +1,82 @@
 //! Contains the full definition of a data system which Ardite will use.
-// TODO: Remove `pub` fields and add a DSL for building definitions.
 
-use linear_map::LinearMap;
+use std::ops::Deref;
 use schema::Schema;
 use value::Key;
 
 /// The definition object which contains all necessary information to
 /// understand an Ardite Schema Definition.
-#[derive(PartialEq, Debug)]
+#[derive(Debug)]
 pub struct Definition {
   /// Types defined in the database.
-  pub types: LinearMap<Key, Type>
+  types: Vec<Type>
+}
+
+impl Definition {
+  /// Creates a new empty instance of `Definition`.
+  pub fn new() -> Self {
+    Definition {
+      types: Vec::new()
+    }
+  }
+
+  /// Add a new type to the `Definition`.
+  pub fn add_type(&mut self, type_: Type) {
+    self.types.push(type_);
+  }
+
+  /// Gets type of a certain name.
+  pub fn find_type<K>(&self, tmp_name: K) -> Option<&Type> where K: Into<Key> {
+    let name = tmp_name.into();
+    self.types.iter().find(|type_| type_.name() == name)
+  }
+}
+
+#[cfg(test)]
+impl PartialEq<Definition> for Definition {
```

Comment:
I'm actually thinking about refactoring the `tests` module soon, making a select group of macros…public…so the `tests` module won't live for much longer.

What I'd really like to be able to do is conditionally `#[derive]` `PartialEq` only in tests and leave the `ParitalEq` `Debug` stuff in the `schema` module.